src/lib.rs

```diff
@@ -221,24 +224,25 @@ impl AstarteSdk {
         Ok(())
     }
 
-    pub async fn add_interface(&self, file_path: &str) -> Result<(), AstarteError> {
+    pub async fn add_interface_from_file(&self, file_path: &str) -> Result<(), AstarteError> {
         let path = Path::new(file_path);
-        match Interface::from_file(path) {
-            Ok(interface) => {
-                if interface.get_ownership() == interface::Ownership::Server {
-                    self.subscribe_server_owned_interface(&interface).await?;
-                }
-                let interfaces_lock = self.interfaces.clone();
-                let mut interfaces = interfaces_lock.write().await;
-                let interfaces_map = &mut interfaces.interfaces;
-                interfaces_map.insert(interface.name().to_string(), interface);
-            }
-            Err(e) => {
-                return Err(AstarteError::BuilderError(
-                    AstarteBuilderError::InterfaceError(e),
-                ))
-            }
+        let interface = Interface::from_file(path)?;
+        self.add_interface(interface).await
+    }
+
+    pub async fn add_interface_from_str(&self, json_str: &str) -> Result<(), AstarteError> {
+        let interface: Interface = Interface::from_str(json_str)?;
+        self.add_interface(interface).await
+    }
+
+    pub async fn add_interface(&self, interface: Interface) -> Result<(), AstarteError> {
+        if interface.get_ownership() == interface::Ownership::Server {
+            self.subscribe_server_owned_interface(&interface).await?;
         }
+        let interfaces_lock = self.interfaces.clone();
+        let mut interfaces = interfaces_lock.write().await;
+        let interfaces_map = &mut interfaces.interfaces;
+        interfaces_map.insert(interface.name().to_string(), interface);
```

Comment:
We should release the rwlock after this line, so that we do not wait for the message to be sent before allowing others (maybe even `send_introspection` itself, check [here](https://github.com/astarte-platform/astarte-device-sdk-rust/blob/5d4ec0a5dc5736382e2f27d53fb1be1f2892f6f8/src/lib.rs#L384)) to access `self.interfaces`.
It would also make possible to defer handling MQTT errors to a later time, using the Astarte introspection synchronization mechanism instead.